src/client/public/scripts/app/app.ts

```diff
@@ -0,0 +1,60 @@
+import './router.js';
+
+/**
+ * Manages pages within the app.
+ */
+class AppElement extends HTMLElement {
+  private modules: string[] = [];
+  private loadedModules: string[] = [];
+
+  connectedCallback() {
+    document.body.addEventListener('url-changed', this.updatePage.bind(this));
+    for (const element of Array.from(this.children)) {
+      this.modules.push(element.id);
+    }
+
+    // Initial load.
+    this.updatePage();
+  }
+
+  /**
+   * Load the correct page based on the path.
+   */
+  updatePage() {
+    const path = window.location.pathname;
+    let module = 'dash';
+    const normalized = path.substring(1);
+    if (this.modules.includes(normalized)) {
```

Comment:
This normalized piece is a little confusing to following, could you add some commebts